becquerel/core/rebin.py

```diff
@@ -291,73 +261,10 @@ def _rebin_listmode(in_spectrum, in_edges, out_edges, slopes):
             in_right_edge - in_left_edge) * np.random.rand(bin_counts)
         energy_idx_start = energy_idx_stop
     # bin the energies (drops the energies outside of the out-binning-range)
-    out_spectrum = np.histogram(energies, bins=out_edges)[0]
-    # add the under/flow counts back into the out_spectrum
-    out_spectrum[0] += (energies < out_edges[0]).nonzero()[0].size
-    out_spectrum[-1] += (energies > out_edges[-1]).nonzero()[0].size
-    return out_spectrum
-
-
-@nb.jit(nb.f8[:, :](nb.f8[:, :], nb.f8[:, :], nb.f8[:], nb.f8[:, :]),
-        locals={'i': nb.u4}, nopython=True)
-def _rebin2d_interpolation(in_spectra, in_edges, out_edges, slopes):
-    """
-    Rebins a 2D array of spectra using linear interpolation
-
-    N.B. Does not keep Poisson statistics
-
-    Wrapper around _rebin_interpolation (1D) for the 2D case
-
-    Args:
-      in_spectra: np.2darray, shape (num_spectra, num_bins_in)
-        array of input spectrum counts of shape
-      in_edges: np.2darray, shape (num_spectra, num_bins_in + 1)
-        array of the input bin edges of shape
-      out_edges: np.1darray
-        array of the output bin edges
-      slopes:
-
-    Returns:
-      np.2darray, shape (num_spectra, num_bins_in)
-      The rebinned spectra
-    """
-    # Init output
-    out_spectra = np.zeros((in_spectra.shape[0], out_edges.shape[0] - 1))
-    for i in np.arange(in_spectra.shape[0]):
-        out_spectra[i, :] = _rebin_interpolation(
-            in_spectra[i, :], in_edges[i, :], out_edges, slopes[i, :])
-    return out_spectra
-
-
-@nb.jit(nb.i8[:, :](nb.i8[:, :], nb.f8[:, :], nb.f8[:], nb.f8[:, :]),
-        locals={'i': nb.u4}, nopython=True)
-def _rebin2d_listmode(in_spectra, in_edges, out_edges, slopes):
-    """
-    Rebins a 2D array of spectra stochastically: histogram to listmode and back
-
-    Wrapper around _rebin_listmode (1D) for the 2D case
-
-    Args:
-      in_spectra: np.2darray, shape (num_spectra, num_bins_in)
-        array of input spectrum counts of shape
-      in_edges: np.2darray, shape (num_spectra, num_bins_in + 1)
-        array of the input bin edges of shape
-      out_edges: np.1darray
-        array of the output bin edges
-      slopes: TODO unused, just for keeping number of arguments the same as
-              _rebin_interpolation()
-
-    Returns:
-      np.2darray, shape (num_spectra, num_bins_in)
-      The rebinned spectra
-    """
-    # Init output
-    out_spectra = np.zeros((in_spectra.shape[0], out_edges.shape[0] - 1),
-                           np.int64)
-    for i in np.arange(in_spectra.shape[0]):
-        out_spectra[i, :] = _rebin_listmode(in_spectra[i, :], in_edges[i, :],
-                                            out_edges, slopes[i, :])
-    return out_spectra
+    # N.B. use [:] to assign values to full array output with guvectorize:
+    out_spectrum[:] = np.histogram(energies, bins=out_edges)[0]
+    # handling of overflows no longer needed since the first/last out_edges
```

Comment:
Move this comment to the docstring

Reply:
Already kind of there. Remove it from here.